src/entry-client.js

```diff
@@ -128,7 +128,9 @@ export default function initializeClient(createApp, clientOpts) {
         router.beforeResolve((to, from, next) => {
             const routeUpdateStr = `${from.fullPath} -> ${to.fullPath}`;
             const fetchDataArgs = { app, route: to, router, store, from };
-            const fetchData = c => isFunction(c.fetchData) && c.fetchData(fetchDataArgs);
```

Comment:
This was problematic because it would resolve to `false` for components that did not specify `fetchData` and cause it to abort the routing operation now that we inspect the results